nrfx/mdk/system_nrf52840.c

```diff
@@ -130,7 +130,21 @@ void SystemInit(void)
             NRF_POWER->RESETREAS =  ~POWER_RESETREAS_RESETPIN_Msk;
         }
     }
-    
+
+    /* Workaround for Errata 197 "DCDC of REG0 not functional"
+     * found at the Errata document for your device located at
+     * https://infocenter.nordicsemi.com/index.jsp
+     */
+    if (nrf52_errata_197()){
+        if (NRF_POWER->DCDCEN0 == 0x1ul)
```

Comment:
`SystemInit()` is not a good place for such thing. The function is called early at the system startup, so it is very likely that `DCDCEN0` will not be configured yet at that point. Instead of in `SystemInit()`, this workaround should be applied later, for example right before DCDC on REG0 is to be actually enabled. I mean something like this:
```
if (nrf52_errata_197()) {
    *(volatile uint32_t *)0x40000638 = 1;
}
NRF_POWER->DCDCEN0 = 1;
```

Reply:
Agree.  That would mean this should be in the zephyr layer at `soc/arm/nordic_nrf/nrf52/soc.c` after
```
#if defined(CONFIG_SOC_DCDC_NRF52X)
	nrf_power_dcdcen_set(true);
#endif
```
correct?

This change could still be useful here if the user is using a bootloader that enabled DCDCEN0 though.  This is the case I ran into.

Reply:
> Agree. That would mean this should be in the zephyr layer at `soc/arm/nordic_nrf/nrf52/soc.c` after
> 
> ```
> #if defined(CONFIG_SOC_DCDC_NRF52X)
> 	nrf_power_dcdcen_set(true);
> #endif
> ```
> 
> correct?

No, `nrf_power_dcdcen_set` sets a different register.
This workaround should be rather placed in the function that modifies `DCDCEN0`, i.e. here:
https://github.com/zephyrproject-rtos/hal_nordic/blob/00580f43b9d48f2079d611ce12f0fbc78d3040f1/nrfx/hal/nrf_power.h#L1150-L1154

> This change could still be useful here if the user is using a bootloader that enabled DCDCEN0 though. This is the case I ran into.

Well, it's actually an issue in this bootloader that activates DCDC on REG0 but does not apply the corresponding workaround, and it should be fixed there.